### PR TITLE
Add development mode to graminize command

### LINKED ISSUE #554: "fx workspace graminize" does not regard the version used to build the workspace
**Describe the bug**

`fx workspace graminize` always uses latest version of OpenFL from Pip, regardless of version used in workspace creation. In case head of develop branch is used, version 1.4 is added to `requirements.txt` and `docker build` fails because there is no such version in Pip repository

**To Reproduce**

```bash
python -m venv venv
. venv/bin/activate
git clone https://github.com/intel/openfl.git
cd openfl
pip install .
fx workspace create --prefix workspace --template torch_unet_kvasir_gramine_ready
cd workspace
fx plan initialize -a `hostname --all-fqdns`
openssl genrsa -3 -out key.pem 3072
fx workspace graminize --rebuild -s key.pem
```

**Expected behavior**

Gramine-based Docker image is created based on the version used to build the workspace (in this case head of develop branch)

### PR DESCRIPTION
**Pull request by itrushkin**
## New Features
 -  `--openfl-path` option to pass to `fx workspace graminize` command. Allows both local and remote OpenFL repository paths.
   1. Local path
   Accepts every path on the system that exists.
   2. Remote path
   Accepts the URL matching repository regular expression. 

 -  New Graminize workflow in GitHub actions (using local repo from the corresponding PR).
   Previously used PyTorch UNet template could not be handled by a GitHub-hosted runner due to memory restrictions. Decreasing of batch size and dataset size does not help, thus I switched the Graminize test to `torch_cnn_histology` template.

closes: #554 #638

### CODE REVIEW

File: openfl/interface/workspace.py [hunk 1]
```diff
@@ -450,9 +450,12 @@ def dockerize_(context, base_image, save):
         default=True, type=bool,
         help='Dump the Docker image to an archive')
 @option('--rebuild', help='Build images with `--no-cache`', is_flag=True)
+@option('--developer-mode', required=False,
+        default=False, type=bool, is_flag=True,
+        help='Use the local OpenFL package in the Docker image')
 @pass_context
 def graminize_(context, signing_key: Path, enclave_size: str, pip_install_options: Tuple[str],
-               save: bool, rebuild: bool) -> None:
+               save: bool, rebuild: bool, developer_mode: bool) -> None:
```
Comment (igor-davidyuk) [inline openfl/interface/workspace.py]: `COPY . .` in the Dockerfile will not work for this command. The reason is `fx workspace graminize` command, as well as `fx workspace dockerize` are designed to be called from an FL workspace, not OpenFL repo.
I imagine the additional parameter to be either a local path to OpenFL repo or a URL for a remote repo.
Something like `openfl_source: Union[Path, str]`, with additional parameter `branch` for the remote option

Comment (itrushkin) [reply openfl/interface/workspace.py]: Got it. We could do this with 1 parameter: the branch could be specified in the URL itself.

**Comment by rahulga1** [comment]
@igor-davidyuk can you please resolve the conflicts and request for review again?

**Comment by MasterSkepticista** [comment]
Gramine-based TEE support has been revamped (#1140), per latest requirements, closing this PR.

**Pull request closed without merging**